tests/system/test_dbapi.py

```diff
@@ -454,6 +456,49 @@ def test_read_timestamp_client_side_autocommit(self):
         read_timestamp_query_result_2 = self._cursor.fetchall()
         assert read_timestamp_query_result_1 != read_timestamp_query_result_2
 
+    @pytest.mark.parametrize("auto_commit", [False, True])
+    def test_batch_ddl(self, auto_commit, dbapi_database):
+        """Test batch ddl."""
+
+        self._conn._buffer_ddl_statements = False
+        if auto_commit:
+            self._conn.autocommit = True
+
+        self._cursor.execute("start batch ddl")
+        self._cursor.execute(
+            """
+            CREATE TABLE Table_1 (
+                SingerId     INT64 NOT NULL,
+                Name    STRING(1024),
+            ) PRIMARY KEY (SingerId)
+            """
+        )
+        self._cursor.execute(
+            """
+            CREATE TABLE Table_2 (
+                SingerId     INT64 NOT NULL,
+                Name    STRING(1024),
+            ) PRIMARY KEY (SingerId)
+            """
+        )
+        exception_raised = False
+        try:
+            self._cursor.execute("run batch")
+        except ProgrammingError:
+            exception_raised = True
+        table_1 = dbapi_database.table("Table_1")
+        table_2 = dbapi_database.table("Table_2")
+        if auto_commit:
+            assert exception_raised is False
+            assert table_1.exists() is True
+            self._cursor.execute("DROP TABLE Table_1")
+            assert table_2.exists() is True
+            self._cursor.execute("DROP TABLE Table_2")
```

Comment:
nit: combine the two `DROP TABLE` statements into one batch, that is more efficient to execute on Spanner. (So first verify that both tables exist, and then drop them)